src/binding/transaction_log_file.cpp

```diff
@@ -187,4 +191,46 @@ void TransactionLogFile::writeEntriesV1(TransactionLogEntryBatch& batch, const u
 		this, bytesWritten, this->size, batch.currentEntryIndex, batch.currentEntryBytesWritten)
 }
 
+/**
+ * Find the position of a timestamp, specifically the position where there are no transactions with an earlier timestamp
+ * @param timestamp - the timestamp to find the position of
+ * @param mapSize - the size of the memory map to search in
+ * @return the position of the timestamp, or zero if comes before this logfile, or 0xFFFFFFFF if it comes after this logfile
+ */
+uint32_t TransactionLogFile::findPositionByTimestamp(double timestamp, uint32_t mapSize) {
+	std::lock_guard<std::mutex> indexLock(this->indexMutex);
+	MemoryMap* memoryMap = getMemoryMap(mapSize);
```

Comment:
nit: I know adding the `this->` is optional, but I found it super helpful to use this convention to quickly identify where to find the method/function.

```suggestion
	MemoryMap* memoryMap = this->getMemoryMap(mapSize);
```